### ISSUE TITLE
Setting the value of Terrain World component property above 1024 leads to a crash

### ISSUE THREAD

**Issue by LB-KacperKapusta** [o3de/o3de#16923]
**Describe the bug**
Setting the value of **Surface Data Query Resolution (m)** property inside the *Terrain World* component property above **1024** leads to a crash when used with *Terrain World Renderer* component. Note that the crash occurs on both Linux and Windows.

Please refer to the attached video and logs for more information.

**Call Stack Trace:**
```
25) 00007FFF6D07E237 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFF6C91C888 (InProcessClient64) : (function-name not available)
23) 00007FFF6C91C4D1 (InProcessClient64) : (function-name not available)
22) 00007FFF6F6954B0 (ntdll) : memset
21) 00007FFF6F67CA06 (ntdll) : _C_specific_handler
20) 00007FFF6F6923AF (ntdll) : _chkstk
19) 00007FFF6F6414B4 (ntdll) : RtlRaiseException
18) 00007FFF6F690EBE (ntdll) : KiUserExceptionDispatcher
17) C:\Builds\o3de\Gems\Terrain\Code\Source\TerrainRenderer\ClipmapBounds.cpp (39) : Terrain::ClipmapBounds::ClipmapBounds
16) C:\Builds\o3de\Gems\Terrain\Code\Source\TerrainRenderer\TerrainDetailMaterialManager.cpp (255) : Terrain::TerrainDetailMaterialManager::InitializeTextureParams
15) C:\Builds\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AzFramework::Terrain::TerrainDataNotifications,AzFramework::Terrain::TerrainDataNotifications,0,1>::Dispatcher<AZ::EBus<AzFramework::Terrain::Te
14) C:\Builds\o3de\Gems\Terrain\Code\Source\TerrainSystem\TerrainSystem.cpp (1829) : Terrain::TerrainSystem::OnTick
13) C:\Builds\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AZ::TickEvents,AZ::TickEvents,0,2>::Dispatcher<AZ::EBus<AZ::TickEvents,AZ::TickEvents> >::Broadcast<void (__cdecl AZ::TickEvents::*)(float,AZ::S
12) C:\Builds\o3de\Code\Framework\AzCore\AzCore\Component\ComponentApplication.cpp (1513) : AZ::ComponentApplication::Tick
11) C:\Builds\o3de\Code\Editor\GameEngine.cpp (823) : CGameEngine::Update
10) C:\Builds\o3de\Code\Editor\CryEdit.cpp (2282) : CCryEditApp::IdleProcessing
 9) C:\Builds\o3de\Code\Editor\CryEdit.cpp (2179) : CCryEditApp::OnIdle
 8) C:\Builds\o3de\Code\Editor\Core\QtEditorApplication.cpp (319) : Editor::EditorQtApplication::maybeProcessIdle
 7) 00007FFF3997F72A (Qt5Core) : QTimer::timeout
 6) 00007FFF399782DD (Qt5Core) : QObject::event
 5) 00007FFF37C849A1 (Qt5Widgets) : QApplicationPrivate::notify_helper
 4) 00007FFF37C83A23 (Qt5Widgets) : QApplication::notify
 3) 00007FFF399517CB (Qt5Core) : QCoreApplication::notifyInternal2
 2) 00007FFF3999C1B9 (Qt5Core) : QEventDispatcherWin32Private::sendTimerEvent
 1) 00007FFF3999A5E9 (Qt5Core) : QEventDispatcherWin32::event
```

**Prerequisites**
* *Terrain* Gem enabled in the project.

**Steps to reproduce**
1. Select the level .prefab entity.
2. Add Terrain World Renderer and Terrain World components to the selected entity.
3. Set value of the **Surface Data Query Resolution (m)** property inside the *Terrain World* component to above 1024.

**Expected behavior**
Editor remains stable.

**Actual behavior**
Editor crashes instantly.

**Screenshots/Video**

https://github.com/o3de/o3de/assets/86953659/84665ba2-b9b6-4e78-8b22-baa1693f0e95

**Found in Branch**
Point-Release/23101
Development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
Point-Release/23101 ([55aa477](https://github.com/o3de/o3de/commit/55aa477c9d6ff2eb0277f9f02a00fa53424b9f7e))
Development ([7291081](https://github.com/o3de/o3de/commit/72910817cc838ecf3aaae1fc0b2448ead354ee7d))

**Desktop/Device**
- Device: PC
- OS: Linux
- Version: Ubuntu 22.04.3 LTS
- CPU: CPU: Intel Core i7-11700F
- GPU: NVIDIA GeForce RTX 3070
- Memory: 32GB

**Additional context:**
[error.log](https://github.com/o3de/o3de/files/13053512/error.log)
[Editor.log](https://github.com/o3de/o3de/files/13053511/Editor.log)
[error.zip](https://github.com/o3de/o3de/files/13053509/error.zip) (containing the generated `error.dmp` file)

**Comment by mbalfour-amzn** [comment]
This specific crash is fixed, though there are other different crashes that can occur still (#16889 for example).

**Comment by LB-KacperKapusta** [comment]
Issue verified as fixed in Point-Release/23101 ([426a5f2](https://github.com/o3de/o3de/commit/426a5f2c4a4312891f20bcc2cf229373307f3204)) branch.
